### PR TITLE
Add keyboard level encoder function

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Added a keyboard level encoder function as many users have complained about losing encoder functionality when using QMK Configurator. 

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/gmmk/pro/pro.c [hunk 1]
```diff
@@ -14,3 +14,15 @@
...
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+  if (!encoder_update_user(index, clockwise)) return false;
+  if (clockwise) {
+    tap_code_delay(KC_VOLU, 10);
+  } else {
+    tap_code_delay(KC_VOLD, 10);
+  }
+  return false;
```
Comment (drashna) [inline keyboards/gmmk/pro/pro.c:26]:
```suggestion
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
    
    if (clockwise) {
        tap_code_delay(KC_VOLU, 10);
    } else {
        tap_code_delay(KC_VOLD, 10);
    }
    return false;
```

**Review by fauxpark — APPROVED** [review]

**Comment by jordysamuel** [comment]
If encoder behavior is already determined in the keymap, `pro.c` interferes with the encoder behavior. I updated my branch to the latest master and my encoder configured for brightness started sending volume up and down as well.

**Comment by gourdo1** [comment]
@drashna Is there a workaround for keymaps that already have advanced encoder functions and layers? As is, this blanket applies an additional volume up/down across all layers.

**Comment by jordysamuel** [comment]
> @drashna @waffle87 Is there a workaround for keymaps that already have advanced encoder functions and layers? As is, this blanket applies an additional volume up/down across all layers.

I just commented out the code in pro.c so it becomes a blank file. For now that seems to be the only way.

**Comment by fauxpark** [comment]
You can just set `ENCODER_ENABLE = no` in your keymap `rules.mk`.

**Comment by tzarc** [comment]
Or, if you still want encoder functionality `return false;` at the end of your keymap's `encoder_update_user()`.

**Comment by gourdo1** [comment]
> Or, if you still want encoder functionality `return false;` at the end of your keymap's `encoder_update_user()`.

Thanks, this worked for me.

**Pull request merged**